src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs

```diff
@@ -229,6 +233,31 @@ internal IChangeToken GetOrAddWildcardChangeToken(string pattern)
             return changeToken;
         }
 
+#if NETSTANDARD2_1
+        public async ValueTask DisposeAsync()
```

Comment:
Another approach to do this, which is what we do in other parts of the repo is declare the class as partial class, then put these two APIs that are included on ns2.1 or later in a separate partial class file and include it in the csproj when target framework is ns2.1 or later.